sno/schema.py

```diff
@@ -303,6 +305,13 @@ def encode_feature_blob(self, feature):
         data = msg_pack([self.legend.hexhash(), non_pk_values])
         return data
 
+    def hash_feature(self, feature):
+        """
+        Given a feature, git-hash it using this schema.
+        Doesn't include its path, so primary key values are not hashed.
```

Comment:
> primary key values are not hashed.

Feels like in general this behaviour is a footgun, right? In the pk-generating scenario it's obviously useful, but it'd be easy to miss it when someone sees there's a `hash_feature()` method. Maybe make it an opt-in behaviour, or rename the method?

Reply:
We use this in two places already - assign PKs, and detect renames - and we don't even need the hash-full-feature version. I think this one is more useful since it does exactly what git does and hashes a file but not the file's name.

However, I see your point that it's not clear from the name or the signature - I'll just add a parameter to hash or not-hash the pk values.